File: diagonalDifference.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the diagonalDifference function below.
# ======================================MY SOLUTION START===========================================
def diagonalDifference(arr):
    n = len(arr) 
    leftVal = 0
    rightVal = 0
    for i in range(n): 
        for j in range(n): 
            # left digonal condition
            if (i == j): 
                leftVal += arr[i][j]
            # light digonal condition
            if(i + j + 1 == n):
                rightVal += arr[i][j]
                print(arr[i][j])
                
    return abs(leftVal - rightVal)
# ======================================MY SOLUTION END===========================================
                
if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input())

    arr = []

    for _ in range(n):
        arr.append(list(map(int, input().rstrip().split())))

    result = diagonalDifference(arr)

    fptr.write(str(result) + '\n')

    fptr.close()

File: diagonalDifference.js
'use strict';

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

// Complete the diagonalDifference function below.
function diagonalDifference(arr) {
    var n = arr.length;
    var i = 0, leftVal = 0, rightVal = 0, leftPointer = 0, rightPointer = n - 1;
    while(i < n){
        // check if length is odd or not, find the midpoint if its odd
        if(n%2 != 0 && leftPointer === rightPointer){
            var midIndex = Math.floor(n%2);
            rightVal += arr[midIndex][midIndex];
            leftVal += arr[midIndex][midIndex];
            leftPointer++, rightPointer--;
        }
        else{
            leftVal += arr[i][leftPointer];
            rightVal += arr[i][rightPointer];
            leftPointer++;
            rightPointer--;
        }
        i++;
    }
    return Math.abs(leftVal - rightVal);
}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    const n = parseInt(readLine(), 10);

    let arr = Array(n);

    for (let i = 0; i < n; i++) {
        arr[i] = readLine().split(' ').map(arrTemp => parseInt(arrTemp, 10));
    }

    const result = diagonalDifference(arr);

    ws.write(result + '\n');

    ws.end();
}

